src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -235,20 +235,38 @@ void MeshForwarder::UpdateIndirectMessages(void)
     }
 }
 
-otError MeshForwarder::EvictIndirectMessage(void)
+otError MeshForwarder::EvictMessage(uint8_t aPriority)
 {
-    otError error = OT_ERROR_NOT_FOUND;
+    otError  error = OT_ERROR_NOT_FOUND;
+    Message *message;
 
-    for (Message *message = mSendQueue.GetHead(); message; message = message->GetNext())
-    {
-        if (!message->IsChildPending())
-        {
-            continue;
-        }
+    VerifyOrExit((message = mSendQueue.GetTail()) != NULL);
 
+    if (message->GetPriority() > aPriority)
+    {
         RemoveMessage(*message);
         ExitNow(error = OT_ERROR_NONE);
     }
+    else
+    {
+        while (1)
+        {
+            for (message = mSendQueue.GetHeadForPriority(aPriority); message; message = message->GetNext())
+            {
+                if (!message->IsChildPending())
+                {
+                    continue;
+                }
+
+                RemoveMessage(*message);
+                ExitNow(error = OT_ERROR_NONE);
+            }
```

Comment:
nit: how about changing the `if` condition and getting rid of `continue` ?
```
if (message->IsChildPending())
{ 
    RemoveMessage(*message);
    ExitNow(error = OT_ERROR_NONE);
}
```

Reply:
Updated.